src/backend/joseki.be/webapp/Handlers/OwnershipExtensions.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using webapp.Database.Cache;
+using webapp.Database.Models;
+
+namespace webapp.Handlers
+{
+    /// <summary>
+    /// IOwnershipCache extension to query component ownership.
+    /// </summary>
+    public static class OwnershipExtensions
+    {
+        /// <summary>
+        /// Returns an owner using a componentId.
+        /// Uses IOwnershipCache to speed up lookup.
+        /// </summary>
+        /// <param name="cache">IOwnershipCache is cached Ownership table.</param>
+        /// <param name="componentId">Id string of the component.</param>
+        /// <returns>String as owner (email).</returns>
+        public static async Task<string> GetOwner(this IOwnershipCache cache, string componentId)
+        {
+            var entries = await cache.GetEntries();
+
+            IComponentId id = ComponentId.ComponentIdFactory(componentId);
+
+            // more detailed component Id should be selected,
+            // for this reason we first query the identity on object level.
+            OwnershipEntity objectOwner = entries.FirstOrDefault(x => x.ComponentId == id.ObjectLevel);
+            if (objectOwner != null && !string.IsNullOrEmpty(objectOwner.Owner))
+            {
+                return objectOwner.Owner;
+            }
+
+            // if no owner on object level returned, check the parent (group owner).
+            OwnershipEntity groupOwner = entries.FirstOrDefault(x => x.ComponentId == id.GroupLevel);
+            if (groupOwner != null && !string.IsNullOrEmpty(groupOwner.Owner))
+            {
+                return groupOwner.Owner;
+            }
+
+            // if no owner on group level returned, check root owner.
+            OwnershipEntity rootOwner = entries.FirstOrDefault(x => x.ComponentId == id.RootLevel);
+            if (rootOwner != null && !string.IsNullOrEmpty(rootOwner.Owner))
+            {
+                return rootOwner.Owner;
+            }
+
+            // if nothing is found, return empty owner.
+            return string.Empty;
+        }
+    }
+#pragma warning disable CS1591
+    /// <summary>
+    /// Component can be either Kubernetes or AzureSubscription.
+    /// </summary>
+    public enum ComponentType
+    {
+        Kubernetes,
+        AzureSubscription,
+    }
+
+    /// <summary>
+    /// Base class for parsing ComponentId.
+    /// </summary>
+    public class ComponentId
```

Comment:
Nice to see this as a separate class finally. I had to do it this way from the very beginning...
What about moving it to a dedicated file?

Reply:
Moved under /Database/Models/ComponentId.cs, looks good?